### PR TITLE
feat(input): new useInputHandle

### PR DESCRIPTION
**Pull request by yqrashawn**
---

## PR Checklist

- [x] Fix linting errors
- [x] Label has been added

## Change information

1. initialValue -> defaultValue
2. remove selfValue in input and textarea
3. add open defaultOpen in auto-complete to enable user to controle visibility of the dropdown
4. add more tests to cover all code

### CODE REVIEW

**Comment by AngeliaQ** [comment]
Task linked: [CU-3xt32w input component: replace useInput, provide both controlled and uncontrolled api](https://app.clickup.com/t/3xt32w)

**Review by GuichiZhao — COMMENTED** [review]
Consider removing useImperativeHandle in `Input`, prefer use `ref` passed from `forwardRef` directly
Consider telling if `Input` in `AutoComplete` using `AutoComplete` context rather than `classNames`

File: pages/en-us/components/input.mdx [hunk 1]
```diff
@@ -330,12 +311,10 @@ type InputSizes = 'mini' | 'small' | 'medium' | 'large'
...
 ```
 
-<Attributes.Title>useInput</Attributes.Title>
+<Attributes.Title>useInputHandle</Attributes.Title>
 
 ```ts
-type useInput = (
-  initialValue: string,
-) => {
+type useInputHandle = () => {
```
Comment (AngeliaQ) [inline pages/en-us/components/input.mdx:317]: please check the useInputHandle api document

**Review by AngeliaQ — CHANGES_REQUESTED** [review]

File: pages/en-us/components/input.mdx (hunk 1 above)
Comment (yqrashawn) [reply pages/en-us/components/input.mdx:317]: done

**Comment by yqrashawn** [comment]
> Consider removing useImperativeHandle in `Input`, prefer use `ref` passed from `forwardRef` directly

I still need to use the inputRef in this component since user may won't pass the ref in.

> Consider telling if `Input` in `AutoComplete` using `AutoComplete` context rather than `classNames`

done

File: components/auto-complete/auto-complete.tsx [hunk 2]
```diff
@@ -71,183 +86,231 @@ const getSearchIcon = (searching?: boolean) => {
...
+              border-bottom-color: transparent;
+            }
+          `}</style>
+        </div>
+      </AutoCompleteContext.Provider>
+    )
+  },
+)
+
+type AutoCompleteComponent<T, P = {}> = React.ForwardRefExoticComponent<
```
Comment (AngeliaQ) [inline components/auto-complete/auto-complete.tsx]: duplicate type define

File: components/auto-complete/index.ts [hunk 3]
```diff
@@ -7,5 +7,7 @@ AutoComplete.Item = AutoCompleteItem
 AutoComplete.Option = AutoCompleteItem
 AutoComplete.Searching = AutoCompleteSearching
 AutoComplete.Empty = AutoCompleteEmpty
+AutoComplete.useAutoCompleteHandle = useAutoCompleteHandle
 
+export { useAutoCompleteHandle }
```
Comment (AngeliaQ) [inline components/auto-complete/index.ts:2]: why export?

File: components/input/__tests__/use-input-handle.test.tsx [hunk 4]
```diff
@@ -0,0 +1,59 @@
+import Input, { useInputHandle } from '../'
+import { mount } from 'enzyme'
+import React, { useEffect } from 'react'
+
+describe('UseImperativeInput', () => {
```
Comment (AngeliaQ) [inline components/input/__tests__/use-input-handle.test.tsx]: may change to UseInputHandle

File: components/input/index.ts [hunk 5]
```diff
@@ -1,8 +1,10 @@
-import Input from './input'
+import Input, { useInputHandle } from './input'
 import Textarea from '../textarea'
 import InputPassword from './password'
 
 Input.Textarea = Textarea
 Input.Password = InputPassword
+Input.useInputHandle = useInputHandle
 
+export { useInputHandle }
```
Comment (AngeliaQ) [inline components/input/index.ts:3]: why export

**Review by AngeliaQ — CHANGES_REQUESTED** [review]

File: components/input/index.ts (hunk 5 above)
Comment (yqrashawn) [reply components/input/index.ts:3]: So that user can `import {useInputHandle} from 'components/input'`

Comment (AngeliaQ) [reply components/input/index.ts:3]: single use Input.useInputHandle as Select.Option? What are the benefits？

Comment (yqrashawn) [reply components/input/index.ts:3]: less code
```javascript
import Input, { useInputHandle } from "components/input"

function () {
  const ... = useInputHandle(...)
  const ... = useInputHandle(...)

  return (
    ...
    <Input id="username" .../>
    <Input id="password" .../>
    ...
  )
}
```

```javascript
import Input from 'components'
const { useInputHandle } = Input

function () {
  const ... = useInputHandle(...)
  const ... = useInputHandle(...)

  return (
    ...
    <Input id="username" .../>
    <Input id="password" .../>
    ...
  )
}
```

```javascript
import Input from 'components'

function () {
  const ... = Input.useInputHandle(...)
  const ... = Input.useInputHandle(...)

  return (
    ...
    <Input id="username" .../>
    <Input id="password" .../>
    ...
  )
}
```

Comment (AngeliaQ) [reply components/input/index.ts:3]: so do you think all the useXXXHandle should provide the use method？

File: components/auto-complete/auto-complete.tsx (hunk 2 above)
Comment (yqrashawn) [reply components/auto-complete/auto-complete.tsx]: done

File: components/auto-complete/index.ts (hunk 3 above)
Comment (yqrashawn) [reply components/auto-complete/index.ts:2]: done

File: components/input/__tests__/use-input-handle.test.tsx (hunk 4 above)
Comment (yqrashawn) [reply components/input/__tests__/use-input-handle.test.tsx]: done

File: components/input/index.ts (hunk 5 above)
Comment (yqrashawn) [reply components/input/index.ts:3]: done

**Review by AngeliaQ — APPROVED** [review]

**Comment by yqrashawn** [comment]
> Consider removing useImperativeHandle in `Input`, prefer use `ref` passed from `forwardRef` directly

I think we need to keep the useInputHandle

> Consider telling if `Input` in `AutoComplete` using `AutoComplete` context rather than `classNames`

done

**Review by GuichiZhao — APPROVED** [review]

**Pull request merged**